engine/src/main/java/org/hibernate/search/engine/backend/types/dsl/ScaledNumberIndexFieldTypeContext.java

```diff
@@ -7,12 +7,24 @@
 package org.hibernate.search.engine.backend.types.dsl;
 
 /**
+ * A context for specifying a scaled number type.
+ *
  * @param <S> The concrete type of this context.
  * @param <F> The type of field values.
  */
 public interface ScaledNumberIndexFieldTypeContext<S extends ScaledNumberIndexFieldTypeContext<? extends S, F>, F>
 		extends StandardIndexFieldTypeContext<S, F> {
 
+	/**
+	 * @param decimalScale How the scale of values should be adjusted before indexing as a fixed-precision integer.
+	 * A positive {@code decimalScale} will shift the decimal point to the right before rounding to the nearest integer and indexing,
+	 * effectively retaining that many after the decimal place in the index,
+	 * but only allowing smaller numbers to be indexed.
```

Comment:
Because larger numbers cannot be indexed anymore. I'll replace with `Since numbers are indexed with a fixed number of bits, this increase in precision also means that the maximum value that can be indexed will be smaller.`